test/lib/jdk/test/lib/hotspot/ir_framework/CompLevel.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.test.lib.hotspot.ir_framework;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Compilation levels used by the framework. The compilation levels map to the used levels in HotSpot (apart from the
+ * framework specific values {@link #SKIP} and {@link #WAIT_FOR_COMPILATION} that cannot be found in HotSpot).
+ *
+ * <p>
+ * The compilation levels can be specified in the {@link Test}, {@link ForceCompile} and {@link DontCompile} annotation.
+ *
+ *
+ * @see Test
+ * @see ForceCompile
+ * @see DontCompile
+ */
+public enum CompLevel {
+    /**
+     * Can only be used at {@link Test#compLevel()}. After the warm-up, the framework keeps invoking the test over a span
+     * of 10s (configurable by setting the property flag {@code -DWaitForCompilationTimeout}) until HotSpot compiles the
+     * {@link Test} method. If the method was not compiled after 10s, an exception is thrown. The framework does not wait
+     * for the compilation if the test VM is run with {@code -Xcomp}, {@code -XX:-UseCompiler} or {@code -DStressCC=true}.
+     */
+    WAIT_FOR_COMPILATION(-4),
+    /**
+     * Can only be used at {@link Test#compLevel()}. Skip a compilation of the {@link Test @Test} method completely.
+     */
+    SKIP(-3),
+    /**
+     *  Use any compilation level depending on the usage:
+     *  <ul>
+     *      <li><p>{@link Test @Test}, {@link ForceCompile @ForceCompile}: Use the highest available compilation level
+     *      which is usually C2.</li>
+     *      <li><p>{@link DontCompile @DontCompile}: Prevents any compilation of the associated helper method.</li>
+     *  </ul>
+     */
+    ANY(-2),
+    /**
+     *  Compilation level 1: C1 compilation without any profile information.
+     */
+    C1(1),
```

Comment:
I initially had `C1_SIMPLE` but then replaced it with `C1` as `@DontCompile` can only exclude a compilation for a compiler and not a specific level. However, while thinking about it, it might be better to not share `C1/C1_SIMPLE` but use a separate annotation for `@DontCompile`. I added a new `Compiler` annotation class and updated the usages and checks.